### PR TITLE
Session

### PR DESCRIPTION
**Pull request by bhflm**
## Summary

Implemented users session using local storage

https://trello.com/c/ETgxfte1/111-routing-i-session

### CODE REVIEW

File: react/tic-tac-toe/src/app/components/AuthenticatedRoutes/index.js [hunk 1]
```diff
@@ -13,22 +12,14 @@ class AuthenticatedRoutes extends Component {
   ];
 
   render() {
-    return this.props.token ? this.routes.map(each => (
+    return window.localStorage.getItem('token') ? this.routes.map(each => (
```
Comment (Dorian30) [inline react/tic-tac-toe/src/app/components/AuthenticatedRoutes/index.js]: I wouldn't handle this here. Why don't you set this as default state in your store, specifically in your token key. You can check whether or not you have something in your localStorage and then set your default state

**Review by Dorian30 — CHANGES_REQUESTED** [review]

File: react/tic-tac-toe/src/redux/users/actions.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const actionsCreators = {
   login: userData => async dispatch => {
     const response = await usersService.userLogin(userData);
     if (response.ok) {
+      window.localStorage.setItem('token', response.data);
```
Comment (Dorian30) [inline react/tic-tac-toe/src/redux/users/actions.js]: You don't need to access window explicitly. All your variables are saved in your window object by default, so JS will always try to find them there. 

```suggestion
      localStorage.setItem('token', response.data);
```

File: react/tic-tac-toe/src/redux/users/reducer.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { actions } from './actions';
 
 const initialState = {
-  token: null
+  token: window.localStorage.getItem('token')
```
Comment (Dorian30) [inline react/tic-tac-toe/src/redux/users/reducer.js]: Same

**Review by Dorian30 — CHANGES_REQUESTED** [review]

**Review by Dorian30 — APPROVED** [review]

File: react/tic-tac-toe/src/redux/users/actions.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ const actionsCreators = {
   login: userData => async dispatch => {
     const response = await usersService.userLogin(userData);
     if (response.ok) {
+      localStorage.setItem('token', response.data);
```
Comment (damfinkel) [inline react/tic-tac-toe/src/redux/users/actions.js:16]: we should be using this to initialize the state, so when the app is reinitialized, the token is taken from local storage and the user is automatically logged in if there's a token there

**Review by damfinkel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**